lib/std/testing.zig

```diff
@@ -199,6 +191,81 @@ test "expectEqual.union(enum)" {
     try expectEqual(a10, a10);
 }
 
+test "expectEqual.optional" {
+    const T = ?u8;
+
+    const a: T = 10;
+    const b: T = null;
+
+    try expectEqual(@as(T, 10), a);
+    try expectEqual(@as(T, null), b);
+}
+
+/// This function is intended to be used only in tests. When the actual is not
+/// an Optional or actual is null or expected is not equal to actual, prints
+/// diagnostics to stderr to show exactly how they are not equal, then returns
+/// a test failure error.
+pub fn expectOptional(expected: anytype, actual: anytype) !void {
+    switch (@typeInfo(@TypeOf(actual))) {
+        .Optional => {
+            if (actual) |actual_payload| {
+                try expectEqual(expected, actual_payload);
+            } else {
+                print("expected {any}, found null\n", .{expected});
+                return error.TestExpectedOptional;
+            }
+        },
+        else => {
+            print("expected type Optional, found type {s}\n", .{@typeName(@TypeOf(actual))});
+            return error.TestExpectedOptional;
+        },
+    }
+}
+
+test "expectOptional" {
```

Comment:
Remove quotes for autodoc to pick it up:  `test expectOptional {` same for expectNull